webapp/channels/src/actions/websocket_actions.jsx

```diff
@@ -733,6 +733,12 @@ export function handleNewPostEvents(queue) {
             console.log('handleNewPostEvents - new posts received', posts);
         }
 
+        posts.forEach((post, index) => {
+            if (queue[index].data.should_ack) {
+                WebSocketClient.acknowledgePostedNotification(post.id, 'not_sent', 'too_many_posts');
```

Comment:
How much work would it be to add a batched version of this API? Given this code path is specifically to avoid sending excess requests to the server, it seems not great to potentially spam the server with WS messages when this triggers, but I don't know if the performance impact of this is enough to worry about that

Reply:
More than I think is worth right now. If the server is capable of sending this many events down without too much trouble, sending a few back for the ones that need to be ACKed shouldn't be too bad either. The only case where this could be a bit troublesome would be if someone was spamming DMs for example, but even then I wouldn't expect that much impact.

I don't think we need to prematurely optimize here, especially since this isn't a case that comes up enough to likely affect much. But we'll know more about that once this change is in.